be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -43,9 +43,10 @@ class RollCallHandler(dbRef: => AskableActorRef) extends MessageHandler {
   def handleCreateRollCall(rpcRequest: JsonRpcRequest): GraphMessage = {
     val ask =
       for {
-        _ <- checkParameters(rpcRequest, serverUnexpectedAnswer)
-        message: Message = rpcRequest.getParamsMessage.get
-        data: CreateRollCall = message.decodedData.get.asInstanceOf[CreateRollCall]
+        (_, message, somedata) <- extractParameters[CreateRollCall](rpcRequest, serverUnexpectedAnswer)
+        data: CreateRollCall = somedata.get
+        // message: Message = rpcRequest.getParamsMessage.get
+        // data: CreateRollCall = message.decodedData.get.asInstanceOf[CreateRollCall]
```

Comment:
if you want me to merge this, remove the old code, don't just comment it 😁 
(this applies throughout the file)

Reply:
this is done in @Ajkunas ' refactoring PR but very good point thank you!